src/App.tsx

```diff
@@ -9,39 +9,30 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { fetchUsers } from './features/userSlice';
+import { fetchPosts, postsSelector } from './features/postSlice';
+import { authorSelector } from './features/authorSlice';
+import { selectedPostSelector } from './features/selectedPostSlice';
+import { set } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const { items: posts, loaded, hasError } = useAppSelector(postsSelector);
+  const author = useAppSelector(authorSelector);
+  const selectedPost = useAppSelector(selectedPostSelector);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  useEffect(() => {
+    dispatch(fetchUsers());
+  }, [dispatch]);
```

Comment:
The `useEffect` hook correctly dispatches `fetchUsers` on component mount. Ensure that the `fetchUsers` action is correctly implemented in the `userSlice` to handle the asynchronous fetching of users .